ci/.travis.yml

```diff
@@ -12,21 +12,14 @@ before_install:
 bundler_args: "--binstubs --standalone --without documentation --path ../bundle"
 script: "script/run_build"
 rvm:
-  - 1.8.7
-  - 1.9.2
-  - 1.9.3
-  - 2.0.0
-  - ree
   - rbx-3
   - jruby-9.1.7.0 # pin JRuby to this until travis/rvm can install later versions
   - jruby-head
-  - jruby-1.7
 env:
   - JRUBY_OPTS='--dev'
+  - RSPEC_CI='true'
 matrix:
   include:
-    - rvm: jruby-1.7
-      env: JRUBY_OPTS='--dev --1.8'
     - rvm: 2.7.1
```

Comment:
You can remove the diff lcs version builds too, and set the minimum diff lcs version.

Reply:
Should I use `">= 1.4.3", "< 2.0"`?
I'll make those changes separately.

Reply:
`>= 1.4.4`

Reply:
Can you just add it as a commit to the prs.

Reply:
Sure, but which one? 😄 I have those four generated from this PR, also those ones from which I've extracted these changes.

I plan to send a few more series:
 - update rake to 13, and cucumber to 2.4 (broke my teeth trying to run 1.3 locally) <--- **will add diff-lcs update here**
 - rethink `define_optimized_require_for_rspec`
 - fix kwargs delegation
 - update JRuby to 9.1.17.0/ add JRuby 9.2
 - get rid of most of the `<<-`/`chomp`/`unindent`/`undent`/`gsub` in favour of `<<~`

Do you have some suggestions for this plan?

Reply:
> update rake to 13, and cucumber to 2.4 (broke my teeth trying to run 1.3 locally) <--- will add diff-lcs update here

Please just add diff-lcs to these PRs, any gem which mentions it in the gemspec just set the minimum version and remove the diff lcs builds.

> *rethink `define_optimized_require_for_rspec`

I want to keep this, it is a lot of churn just to remove a generated method.

> fix kwargs delegation

We may find this is impossible due to the hash behaviour, I suspect we will need to keep ruby 2 keywords for quite some time.

> get rid of most of the <<-/chomp/unindent/undent/gsub in favour of <<~

That can wait until later prs, its subjective please don't combine it into your mega PRs they are already too hard to review.

> update JRuby to 9.1.17.0/ add JRuby 9.2

Do this seperatly as I'm also working on this for main

Reply:
> I plan to send a few more series

I'd like to emphasize and elaborate on the above statement ☝️ to remove all the ambiguity.
I plan to send series (dev+support+core+mocks+expectations) of PRs for each of those changes separately.
After this current one is merged.
All against `4-0-dev`.

>> define_optimized_require_for_rspec

> I want to keep this

👍 

> Please just add diff-lcs to these PRs, any gem which mentions it in the gemspec just set the minimum version and remove the diff lcs builds.

Will do in a follow-up 👍 

>> fix kwargs delegation

> this is impossible due to the hash behaviour, I suspect we will need to keep ruby 2 keywords for quite some time.

That's is the plan, to use `ruby2_keywords`, there's no other way around. I intend to use [this comment](https://github.com/rspec/rspec-expectations/issues/1221#issuecomment-716133588) as the guideline.